openmdao/recorders/base_recorder.py

```diff
@@ -0,0 +1,294 @@
+                             desc='Set to True to record responses at the driver level')
+        self.options.declare('record_objectives', type_=bool, default=False,
+                             desc='Set to True to record objectives at the driver level')
+        self.options.declare('record_constraints', type_=bool, default=False,
+                             desc='Set to True to record constraints at the driver level')
+        # Solver options
+        self.options.declare('record_abs_error', type_=bool, default=True,
+                             desc='Set to True to record absolute error at the solver level')
+        self.options.declare('record_rel_error', type_=bool, default=True,
+                             desc='Set to True to record relative error at the solver level')
+        self.options.declare('record_solver_output', type_=bool, default=False,
+                             desc='Set to True to record output at the solver level')
+        self.options.declare('record_solver_residuals', type_=bool, default=False,
+                             desc='Set to True to record residuals at the solver level')
+
+        self.out = None
+
+        # global counter that is used in iteration coordinate
+        self._counter = 0
+
+        # dicts in which to keep the included items for recording
+        self._filtered_driver = {}
+        self._filtered_system = {}
+        self._filtered_solver = {}
+
+    def startup(self, object_requesting_recording):
+        """
+        Prepare for a new run and calculate inclusion lists.
+
+        Parameters
+        ----------
+        object_requesting_recording :
+            Object to which this recorder is attached.
+        """
+        self._counter = 0
+
+        # Deprecated options here, but need to preserve backward compatibility if possible.
+        if self.options['record_params']:
+            warn_deprecation("record_params is deprecated, please use record_inputs.")
+            # set option to what the user intended.
+            self.options['record_inputs'] = True
+
+        if self.options['record_unknowns']:
+            warn_deprecation("record_ is deprecated, please use record_inputs.")
+            # set option to what the user intended.
+            self.options['record_outputs'] = True
+
+        if self.options['record_resids']:
+            warn_deprecation("record_params is deprecated, please use record_inputs.")
+            # set option to what the user intended.
+            self.options['record_residuals'] = True
+
+        # Compute the inclusion/exclusion lists
+
+        if (isinstance(object_requesting_recording, System)):
+            myinputs = myoutputs = myresiduals = set()
+            incl = self.options['includes']
+            excl = self.options['excludes']
+
+            if self.options['record_inputs']:
+                if object_requesting_recording._inputs:
+                    myinputs = [n for n in object_requesting_recording._inputs._names
```

Comment:
That is almost certainly just a mistake, don't remember a reason to convert.  Thanks.